exp/controllers/azuremanagedcluster_controller.go

```diff
@@ -57,10 +57,22 @@ type AzureManagedClusterReconciler struct {
 func (r *AzureManagedClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
 	log := r.Log.WithValues("controller", "AzureManagedCluster")
 	azManagedCluster := &infrav1exp.AzureManagedCluster{}
+
+	// create mapper to transform incoming AzureManagedControlPlanes into AzureManagedCluster requests
+	azureManagedControlPlaneMapper, err := AzureManagedControlPlaneToAzureManagedClusterMapper(ctx, r.Client, log)
+	if err != nil {
+		return errors.Wrap(err, "failed to create AzureManagedControlPlane to AzureManagedClusters mapper")
+	}
+
 	c, err := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
 		For(azManagedCluster).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
+		// watch AzureManagedControlPlane resources
+		Watches(
+			&source.Kind{Type: &infrav1exp.AzureManagedControlPlane{}},
+			handler.EnqueueRequestsFromMapFunc(azureManagedControlPlaneMapper),
```

Comment:
this creates a circular watch from AMCP -> AMC -> AMCP but it terminates because the patches will be no-ops after the first loop (which is necessary to get the ownerRef on AMC).